packages/mobile/src/services/offline-downloader/workers/trackDownloadWorker.ts

```diff
@@ -1,17 +1,43 @@
 import type { CancellablePromise } from 'react-native-job-queue'
-import { Worker } from 'react-native-job-queue'
+import queue, { Worker } from 'react-native-job-queue'
 
+import { isEqualTrackPayload } from 'app/services/offline-downloader/offline-download-queue'
 import { store } from 'app/store'
 import { errorDownload } from 'app/store/offline-downloads/slice'
 
-import { batchRemoveTrackDownload, downloadTrack } from '../offline-downloader'
+import {
+  batchRemoveTrackDownload,
+  downloadTrack,
+  DownloadTrackError
+} from '../offline-downloader'
 import type { TrackForDownload } from '../types'
 
 export const TRACK_DOWNLOAD_WORKER = 'track_download_worker'
 export type TrackDownloadWorkerPayload = TrackForDownload
 
-const onFailure = ({ payload }: { payload: TrackForDownload }) => {
-  store.dispatch(errorDownload(payload.trackId.toString()))
+const onFailure = async (
+  { payload }: { payload: TrackForDownload },
+  error: Error
+) => {
+  switch (error.message) {
+    case DownloadTrackError.IS_DELETED:
+    case DownloadTrackError.IS_UNLISTED: {
```

Comment:
and same for here for tracks, really nice sleuthing with this lib @raymondjacobson!